homeassistant/components/risco/__init__.py

```diff
@@ -81,15 +83,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def _async_setup_local_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     data = entry.data
-    risco = RiscoLocal(data[CONF_HOST], data[CONF_PORT], data[CONF_PIN])
+    comm_delay = initial_delay = data.get(CONF_COMMUNICATION_DELAY, 0)
 
-    try:
-        await risco.connect()
-    except CannotConnectError as error:
-        raise ConfigEntryNotReady() from error
-    except UnauthorizedError:
-        _LOGGER.exception("Failed to login to Risco cloud")
-        return False
+    while True:
+        try:
+            risco = RiscoLocal(
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data[CONF_PIN],
+                **{CONF_COMMUNICATION_DELAY: comm_delay},
```

Comment:
I'd just use a keyword argument directly instead of creating a dict and unpacking it inline.